pkg/model/message.go

```diff
@@ -14,6 +14,36 @@ type Messages struct {
 	Original bool
 }
 
+type MessagesSlice []Messages
+
+// Clone() returns a deep copy of MessagesSlice.
+func (m MessagesSlice) Clone() MessagesSlice {
+	msgs := make(MessagesSlice, len(m))
```

Comment:
Let's return a nil slice instead of an empty slice when `len(m) == 0`.

```suggestion
	msgs := make(MessagesSlice, len(m))
```
```suggestion
	if len(m) == 0 {
		nil
	}

	msgs := make(MessagesSlice, len(m))
```